src/multizimbutton.cpp

```diff
@@ -44,12 +44,8 @@ void MultiZimButton::update_display()
         item->setData(Qt::UserRole, bookId);
         item->setData(Qt::DisplayRole, bookTitle);
 
-        const auto radioBt = new QRadioButton(bookTitle);
-        radioBt->setIcon(zimIcon);
-        mp_radioButtonGroup->addButton(radioBt);
-
         mp_buttonList->addItem(item);
-        mp_buttonList->setItemWidget(item, radioBt);
+        setItemZimWidget(item, bookTitle, zimIcon);
```

Comment:
As of this commit `setItemZimWidget()` looks artificial. I would replace it with a fatter function `MultiZimButton::addBook(QString bookId)` and maybe move all code in the loop body into it. But let's see if further evolution of this PR justifies your implementation.

Reply:
I see, your way of bringing the current ZIM to top stands in the way of such proposal.